libcodechecker/analyze/log_parser.py

```diff
@@ -277,12 +276,31 @@ def parse_compile_commands_json(logfile, parseLogOptions):
                 if not os.path.isdir(inc_dir):
                     compile_opts[i] = '-I' + \
                         os.path.join(entry['directory'], inc_dir)
+
         action.analyzer_options = compile_opts
 
         action.lang = results.lang
         action.target = results.arch
         action.output = results.output
 
+        add_compiler_defaults = True
+
+        # With gcc-toolchain a non default compiler toolchain can be set.
+        # Clang will search for include paths and libraries based on the
+        # gcc-toolchain parameter.
+        # Detecting extra include paths from the host compiler could
+        # conflict with this.
+
+        # For example if the compiler in the compile command is clang
+        # and gcc-toolchain is set we will get the include paths
+        # for clang and not for the compiler set in gcc-toolchain.
+        # This can cause missing headers during the analysis.
+
+        for compiler_option in action.analyzer_options:
+            if '--gcc-toolchain' in compiler_option:
+                add_compiler_defaults = False
```

Comment:
If `add_compiler_defaults` is `False` then `get_compiler_includes` and `get_compiler_target` is never invoked. And that is a problem. Those functions create two additional json files which are essential in the failure zip; we use them to reproduce any failure locally.

Couldn't we just swallow the `--gcc-toolchain` param?